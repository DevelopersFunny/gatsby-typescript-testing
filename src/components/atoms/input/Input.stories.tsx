import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { userEvent, within } from '@storybook/testing-library'
import { Input } from '.'

export default {
  title: 'Atoms/Input',
  component: Input,
  parameters: {
    fileName: __filename,
  },
} as ComponentMeta<typeof Input>

const Template: ComponentStory<typeof Input> = args => <Input {...args} />

/** */
export const InputArgsRequired = Template.bind({})
InputArgsRequired.args = {
  type: 'text',
  id: 'name',
  name: 'name',
  variant: 'standard',
  placeholder: 'Please write a placeholder',
}
/** */
export const InputOutlined = Template.bind({})
InputOutlined.args = {
  type: 'text',
  id: 'name',
  name: 'name',
  placeholder: 'Please write a placeholder',
  variant: 'outlined',
}
/** */
export const InputFilled = Template.bind({})
InputFilled.args = {
  type: 'text',
  id: 'name',
  name: 'name',
  placeholder: 'Please write a placeholder',
  variant: 'filled',
}
/** */
export const InputStandard = Template.bind({})
InputStandard.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await userEvent.type(canvas.getByTestId('name'), '1h2y3n4u', {
    delay: 100,
  })
}
InputStandard.args = {
  type: 'text',
  id: 'name',
  name: 'name',
  placeholder: 'Please write a placeholder',
  variant: 'standard',
  minLength: 4,
  maxLength: 8,
}
/** */
export const InputNumber = Template.bind({})
InputNumber.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await userEvent.type(canvas.getByTestId('number'), '150', {
    delay: 100,
  })
}
InputNumber.args = {
  type: 'number',
  id: 'number',
  name: 'number',
  variant: 'standard',
}
/** */
export const InputPassword = Template.bind({})
InputPassword.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await userEvent.type(canvas.getByTestId('password'), 'supersecret', {
    delay: 100,
  })
}
InputPassword.args = {
  type: 'password',
  id: 'password',
  name: 'password',
  placeholder: 'Password',
}
/** */
export const InputFile = Template.bind({})
InputFile.args = {
  type: 'file',
  id: 'file',
  name: 'file',
  variant: 'standard',
}
/** */
export const InputRange = Template.bind({})
InputRange.args = {
  type: 'range',
  id: 'range',
  name: 'range',
  variant: 'standard',
}
/** */
export const InputRadio = Template.bind({})
InputRadio.args = {
  type: 'radio',
  id: 'radio',
  name: 'radio',
  variant: 'standard',
}
/** */
export const InputCheckbox = Template.bind({})
InputCheckbox.args = {
  type: 'checkbox',
  id: 'checkbox',
  name: 'checkbox',
  variant: 'standard',
}
