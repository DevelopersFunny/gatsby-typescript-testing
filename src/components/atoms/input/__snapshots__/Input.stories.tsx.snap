// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories storyshots Atoms/Input Input Args Required 1`] = `
<input
  className="css-16xa3tr"
  data-testid="name"
  id="name"
  name="name"
  placeholder="Please write a placeholder"
  type="text"
/>
`;

exports[`Stories storyshots Atoms/Input Input Checkbox 1`] = `
<input
  className="css-16xa3tr"
  data-testid="checkbox"
  id="checkbox"
  name="checkbox"
  type="checkbox"
/>
`;

exports[`Stories storyshots Atoms/Input Input File 1`] = `
<input
  className="css-44zl3k"
  data-testid="file"
  id="file"
  name="file"
  type="file"
/>
`;

exports[`Stories storyshots Atoms/Input Input Filled 1`] = `
<input
  className="css-p4oc1c"
  data-testid="name"
  id="name"
  name="name"
  placeholder="Please write a placeholder"
  type="text"
/>
`;

exports[`Stories storyshots Atoms/Input Input Number 1`] = `
<input
  className="css-16xa3tr"
  data-testid="number"
  id="number"
  name="number"
  type="number"
/>
`;

exports[`Stories storyshots Atoms/Input Input Outlined 1`] = `
<input
  className="css-6ui80c"
  data-testid="name"
  id="name"
  name="name"
  placeholder="Please write a placeholder"
  type="text"
/>
`;

exports[`Stories storyshots Atoms/Input Input Password 1`] = `
<input
  className="css-1xsr6mf"
  data-testid="password"
  id="password"
  name="password"
  placeholder="Password"
  type="password"
/>
`;

exports[`Stories storyshots Atoms/Input Input Radio 1`] = `
<input
  className="css-16xa3tr"
  data-testid="radio"
  id="radio"
  name="radio"
  type="radio"
/>
`;

exports[`Stories storyshots Atoms/Input Input Range 1`] = `
<input
  className="css-16xa3tr"
  data-testid="range"
  id="range"
  name="range"
  type="range"
/>
`;

exports[`Stories storyshots Atoms/Input Input Standard 1`] = `
<input
  className="css-16xa3tr"
  data-testid="name"
  id="name"
  maxLength={8}
  minLength={4}
  name="name"
  placeholder="Please write a placeholder"
  type="text"
/>
`;
